cpp/src/search/contains_nested.cu

```diff
@@ -59,4 +65,87 @@ bool contains_nested_element(column_view const& haystack,
   });
 }
 
+std::unique_ptr<column> multi_contains_nested_elements(column_view const& haystack,
+                                                       column_view const& needles,
+                                                       rmm::cuda_stream_view stream,
+                                                       rmm::mr::device_memory_resource* mr)
+{
+  auto result = make_numeric_column(data_type{type_to_id<bool>()},
+                                    needles.size(),
+                                    copy_bitmask(needles),
+                                    needles.null_count(),
+                                    stream,
+                                    mr);
+  if (needles.is_empty()) { return result; }
+
+  auto const out_begin = result->mutable_view().template begin<bool>();
+  if (haystack.is_empty()) {
+    thrust::uninitialized_fill(
+      rmm::exec_policy(stream), out_begin, out_begin + needles.size(), false);
+    return result;
+  }
+
+  auto const haystack_tv        = table_view{{haystack}};
+  auto const needles_tv         = table_view{{needles}};
+  auto const haystack_has_nulls = has_nested_nulls(haystack_tv);
+  auto const needles_has_nulls  = has_nested_nulls(needles_tv);
+
+  auto haystack_map =
+    detail::hash_map_type{compute_hash_table_size(haystack.size()),
+                          detail::COMPACTION_EMPTY_KEY_SENTINEL,
+                          detail::COMPACTION_EMPTY_VALUE_SENTINEL,
+                          detail::hash_table_allocator_type{default_allocator<char>{}, stream},
+                          stream.value()};
+
+  // Insert all indices of the elements in the haystack column into the hash map.
+  // As such, we will use `thrust::equal_to` as key comparator to not ignore any key.
+  //
+  // An alternative way to this is to use `self_comparator` for key comparisons, which would only
+  // insert unique rows of the haystack column. This would save some memory (or significant amount
+  // of memory if the haystack column contains many duplicate rows), however, will result in much
+  // more processing time due to expensive row comparisons.
+  {
```

Comment:
Are these braces meant to constrain the lifetimes of the hashers?

Reply:
For two reasons:
 1. To separate steps.
 2. To reuse variables `hasher` and `comparator`. Otherwise I have to have 2 separate `_hasher` and `_comparator`.